core/cmd/shell.go

```diff
@@ -97,12 +98,30 @@ func initGlobals(cfgProm config.Prometheus, cfgTracing config.Tracing, cfgTeleme
 			for k, v := range cfgTelemetry.ResourceAttributes() {
 				attributes = append(attributes, attribute.String(k, v))
 			}
+
+			keys, err := ksCSA.GetAll()
+			if err != nil {
+				return err
+			}
+			if len(keys) < 1 {
+				return errors.New("Could not bootstrap Beholder, CSA key does not exist")
+			}
+
+			key := keys[0]
+			csaPubKey := key.PublicKey
+			csaPrivKey := key.Raw().Bytes()
+			csaSigner := func(data []byte) []byte {
+				return ed25519.Sign(csaPrivKey, data)
+			}
```

Comment:
How about pre-calculating this and passing the key&signer as args to `initGlobals`?

Reply:
This also makes it easy to pass the values to `plugins.NewLoopRegistry`